inception-ui-external-search/src/main/java/de/tudarmstadt/ukp/inception/app/ui/externalsearch/sidebar/ExternalSearchAnnotationSidebar.java

```diff
@@ -0,0 +1,427 @@
+                .existsSourceDocument(project, documentTitle);
+
+            // Import and open annotation
+            LambdaAjaxLink link;
+            if (!existsSourceDocument) {
+                link = new LambdaAjaxLink("docLink", _target -> {
+                    selectedResult = result;
+                    try {
+                        documentImporter
+                            .importDocumentFromDocumentRepository(result.getDocumentTitle(),
+                                searchStateModel.getObject().getCurrentRepository());
+
+                        getAnnotationPage().actionShowSelectedDocument(_target,
+                            documentService.getSourceDocument(project, documentTitle));
+                    }
+                    catch (IOException e) {
+                        LOG.error(e.getMessage(), e);
+                        error(e.getMessage() + " - " + ExceptionUtils.getRootCauseMessage(e));
+                        e.printStackTrace();
+                    }
+                });
+            }
+            else {
+                // open action
+                link = new LambdaAjaxLink("docLink", _target -> {
+                    selectedResult = result;
+                    getAnnotationPage().actionShowSelectedDocument(_target,
+                        documentService.getSourceDocument(project, documentTitle));
+                });
+            }
+
+            String title = defaultIfBlank(result.getDocumentTitle(),
+                defaultIfBlank(result.getDocumentId(),
+                    defaultIfBlank(result.getUri(), "<no title>")));
+
+            add(link);
+
+            link.add(new Label("title", title));
+            link.add(new Label("score", result.getScore()));
+            link.add(new Label("highlight", highlight).setEscapeModelStrings(false));
+            link.add(new Label("importStatus",
+                () -> existsSourceDocument ? "imported" : "not imported"));
+        }
+    }
+
+    public static class ExternalSearchUserState
+        implements Serializable
+    {
+        private AnnotationLayer layer;
+
+        private DocumentRepository currentRepository = null;
+
+        private IModel<String> targetQuery = null;
+
+        private ExternalResultDataProvider dataProvider = null;
+
+        private long currentPage = 1;
+
+        public DocumentRepository getCurrentRepository()
+        {
+            return currentRepository;
+        }
+
+        public void setCurrentRepository(DocumentRepository repository)
+        {
+            this.currentRepository = repository;
+        }
+
+        public AnnotationLayer getLayer()
+        {
+            return layer;
+        }
+
+        public void setLayer(AnnotationLayer selectedLayer)
+        {
+            this.layer = selectedLayer;
+        }
+
+        public IModel<String> getTargetQuery()
+        {
+            return targetQuery;
+        }
+
+        public void setTargetQuery(IModel<String> query)
```

Comment:
I assume you call it `targetQuery` because we used that name in other parts. Honestly, I never understood why it wasn't simply `query`. If you as me: rename it.

Reply:
Renamed to `query`